VirtoCommerce.CatalogModule.Data/Services/CategoryMover.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections.Generic;
+using System.Linq;
+
+using VirtoCommerce.CatalogModule.Web.Model;
+using VirtoCommerce.CatalogModule.Web.Services;
+using VirtoCommerce.Domain.Catalog.Services;
+using VirtoCommerce.Platform.Core.Common;
+
+using VC = VirtoCommerce.Domain.Catalog.Model;
+
+namespace VirtoCommerce.CatalogModule.Data.Services
+{
+    public class CategoryMover : ListEntryMover<VC.Category>
+    {
+        private readonly ICategoryService _categoryService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CategoryMover"/> class.
+        /// </summary>
+        /// <param name="categoryService">
+        /// The category service.
+        /// </param>
+        public CategoryMover(ICategoryService categoryService)
+        {
+            _categoryService = categoryService;
+        }
+
+        public override void ConfirmMove(IEnumerable<VC.Category> entities)
+        {
+            _categoryService.Update(entities.ToArray());
+        }
+
+        public override List<VC.Category> PrepareMove(MoveContext moveContext)
+        {
+            var result = new List<VC.Category>();
+
+            foreach (var listEntryCategory in moveContext.ListEntries.Where(
```

Comment:
No need to break the line, it is just one pretty short LINQ statement (less than 200 symbols).